packages/dds/map/src/directory.ts

```diff
@@ -697,10 +700,14 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         this.messageHandlers.set(
             "clear",
             {
-                process: (op: IDirectoryClearOperation, local, localOpMetadata) => {
+                process: (msg: ISequencedDocumentMessage, op: IDirectoryClearOperation, local, localOpMetadata) => {
                     const subdir = this.getWorkingDirectory(op.path) as SubDirectory | undefined;
                     if (subdir) {
-                        subdir.processClearMessage(op, local, localOpMetadata);
+                        if(subdir.clientIds.includes(msg.clientId)
+                        || (subdir.sequenceNumber !== -1
+                            && msg.referenceSequenceNumber <= subdir.sequenceNumber)){
```

Comment:
besides creators, only clients whos ref seq is at or above the dir seq can see it. the dirs seq will change only if it gets deleted and recreated